src/buffer/out/textBuffer.hpp

```diff
@@ -229,33 +229,40 @@ class TextBuffer final
     std::wstring GetCustomIdFromId(uint16_t id) const;
     void CopyHyperlinkMaps(const TextBuffer& OtherBuffer);
 
-    class TextAndColor
-    {
-    public:
-        std::vector<std::wstring> text;
-        std::vector<std::vector<COLORREF>> FgAttr;
-        std::vector<std::vector<COLORREF>> BkAttr;
-    };
-
     size_t SpanLength(const til::point coordStart, const til::point coordEnd) const;
 
-    const TextAndColor GetText(const bool includeCRLF,
-                               const bool trimTrailingWhitespace,
-                               const std::vector<til::inclusive_rect>& textRects,
-                               std::function<std::pair<COLORREF, COLORREF>(const TextAttribute&)> GetAttributeColors = nullptr,
-                               const bool formatWrappedRows = false) const;
+    std::vector<std::wstring> GetText(const std::vector<til::inclusive_rect>& selectionRects,

```

Comment:
I kept `GetText` and also made `RetrieveSelectedTextFromBufferRows` which calls `GetText`, because `ControlCore::SelectedText` has a hard dependency on the calls it makes to get selected text to return a vector of strings.

Maybe If I can understand how winrt side of things works, I should be able to refactor SelectedText as well.

Reply:
`TextBuffer::GetText()` is removed.